### PR TITLE
fix: crash when bad network

### PR DESCRIPTION
**Pull request by alantoa**
# Why

Using the app with a very bad network resulted in a forced logout.

# How

- add a patch file for web3-provider-engine  lib to avoid crashes when bad network.
- if no network connection or auth fails will not log out and use cache data on App.
- use snackbar to show `No internet connection` hints when device does not network.

# Test Plan

- check if it crashes when a bad network is on App.

### CODE REVIEW

File: packages/app/hooks/use-network-connection.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const { bottom } = useSafeAreaInsets();
+  const isForeground = useIsForeground();
+  const snackbar = useSnackbar();
+
+  useEffect(() => {
+    const getNetwork = async () => {
+      const { isConnected } = await Network.getNetworkStateAsync();
+      if (isConnected) return;
+      snackbar?.show({
+        text: `No internet connection`,
```
Comment (axeldelafosse) [inline packages/app/hooks/use-network-connection.tsx:20]: 💯

File: patches/eth-block-tracker+4.4.3.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+-          this.emit('error', newErr)
+-        } catch (emitErr) {
+-          console.error(newErr)
+-        }
++        // try {
++        //   this.emit('error', newErr)
++        // } catch (emitErr) {
++        //   console.error(newErr)
++        // }
++        // avoid this.emit('error', newErr) to cause crash on native
```
Comment (axeldelafosse) [inline patches/eth-block-tracker+4.4.3.patch:19]: Nice catch!

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM ❇️

**Pull request merged**